pkg/api/norman/customization/aks/listers.go

```diff
@@ -0,0 +1,234 @@
+	}
+
+	clientContainer, err := NewContainerServiceClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	orchestrators, err := clientContainer.ListOrchestrators(ctx, cap.ResourceLocation, "managedClusters")
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get orchestrators: %v", err)
+	}
+
+	if orchestrators.Orchestrators == nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("no version profiles returned: %v", err)
+	}
+
+	var kubernetesVersions []string
+
+	for _, profile := range *orchestrators.Orchestrators {
+		if profile.OrchestratorType == nil || profile.OrchestratorVersion == nil {
+			return nil, http.StatusInternalServerError, fmt.Errorf("unexpected nil orchestrator type or version")
+		}
+
+		if *profile.OrchestratorType == "Kubernetes" {
+			kubernetesVersions = append(kubernetesVersions, *profile.OrchestratorVersion)
+		}
+	}
+
+	sort.Sort(sortableVersion(kubernetesVersions))
+
+	return encodeOutput(kubernetesVersions)
+}
+
+func listVirtualNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	if cap.TenantID == "" {
+		return nil, http.StatusBadRequest, fmt.Errorf("tenantId is required")
+	}
+
+	clientNetwork, err := NewNetworkServiceClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	result, err := clientNetwork.ListAll(ctx)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get networks: %v", err)
+	}
+
+	var networks []virtualNetworksResponseBody
+
+	for result.NotDone() {
+		var batch []virtualNetworksResponseBody
+
+		for _, azureNetwork := range result.Values() {
+			var subnets []subnet
+
+			if azureNetwork.Subnets != nil {
+				for _, azureSubnet := range *azureNetwork.Subnets {
+					if azureSubnet.Name != nil {
+						subnets = append(subnets, subnet{
+							Name:         to.String(azureSubnet.Name),
+							AddressRange: to.String(azureSubnet.AddressPrefix),
+						})
+					}
+				}
+			}
+
+			if azureNetwork.ID == nil {
+				return nil, http.StatusInternalServerError, fmt.Errorf("no ID on virtual network")
+			}
+
+			match := matchAzureNetwork.FindStringSubmatch(*azureNetwork.ID)
+
+			if len(match) < 2 || match[1] == "" {
+				return nil, http.StatusInternalServerError, fmt.Errorf("could not parse virtual network ID")
+			}
+
+			if azureNetwork.Name == nil {
+				return nil, http.StatusInternalServerError, fmt.Errorf("no name on virtual network")
+			}
+
+			batch = append(batch, virtualNetworksResponseBody{
+				Name:          to.String(azureNetwork.Name),
+				ResourceGroup: match[1],
+				Subnets:       subnets,
+			})
+		}
+
+		networks = append(networks, batch...)
+
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			return nil, http.StatusInternalServerError, err
+		}
+	}
+
+	return encodeOutput(networks)
+}
+
+type locationsResponseBody struct {
+	Name        string `json:"name"`
+	DisplayName string `json:"displayName"`
+}
```

Comment:
Can we use a `map` like
`{
	"name": "displayName"
}`
instead of creating a new type?

Reply:
Changing it will delay UI implementation, I would keep it as it is

Reply:
Why does it affect UI? In either case it's marshalled to JSON.

Reply:
It will have different structure format, currently we have
```
[{"name":"eastasia","displayName":"East Asia"},{"name":"southeastasia","displayName":"Southeast Asia"}]
```
and you want to have
```
[{"eastasia":"East Asia"},{""southeastasia":"Southeast Asia"}]
```

For me this change is ok, but it will brake current UI integration. At the same time this should be adopted on UI site.